packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -152,7 +154,96 @@ export class TabBarRenderer extends TabBar.Renderer {
             top = `${paddingTop + iconHeight}px`;
         }
         const style: ElementInlineStyle = { width, height, top };
-        return h.div({ className: 'p-TabBar-tabLabel', style }, data.title.label);
+        if (isInSidePanel || (this.tabBar && this.tabBar.titles.length < 2)) {
+            return h.div({ className: 'p-TabBar-tabLabel', style }, data.title.label);
+        }
+        const originalToDisplayedMap = this.findDuplicateLabels();
+        const labelDetails: string | undefined = originalToDisplayedMap.get(data.title.caption);
+        return (labelDetails)
```

Comment:
Can we refactor this part of the method for readability? (perhaps simply use `if` statements)

Reply:
Ok sure!